tests/base_test.go

```diff
@@ -0,0 +1,606 @@
+	}
+
+	plan := &v1beta1.Plan{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "Plan",
+			APIVersion: "forklift.konveyor.io/v1beta1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "plan-test",
+			Namespace: namespaceName,
+		},
+		Spec: v1beta1.PlanSpec{
+			Provider: provider.Pair{
+				Destination: corev1.ObjectReference{
+					Name:      "host",
+					Namespace: providerName.Namespace,
+				},
+				Source: corev1.ObjectReference{
+					Name:      providerIdentifier.Name,
+					Namespace: providerIdentifier.Namespace,
+				},
+			},
+			Archived:        false,
+			Warm:            false,
+			TargetNamespace: providerIdentifier.Namespace,
+			Map: plan.Map{
+				Storage: corev1.ObjectReference{
+					Name:      "storage-map-test",
+					Namespace: namespaceName,
+				},
+				Network: corev1.ObjectReference{
+					Name:      "network-map-test",
+					Namespace: namespaceName,
+				},
+			},
+			VMs: []plan.VM{
+				{
+					Ref: ref.Ref{
+						ID: vm.MustId(),
+					},
+				},
+			},
+		},
+	}
+
+	log.Info("Creating Plan...")
+
+	defer cl.Delete(context.TODO(), plan, &client.DeleteOptions{})
+
+	err = cl.Create(context.TODO(), plan, &client.CreateOptions{})
+	if err != nil {
+		t.Fatal(err, "Failed to create the plan")
+	}
+	done = make(chan struct{})
+
+	returnedPlan := &v1beta1.Plan{}
+	planIdentifier := types.NamespacedName{Namespace: plan.Namespace, Name: plan.Name}
+	// Wait for provider to be ready
+	statusCheck = func() {
+		err = cl.Get(context.TODO(), planIdentifier, returnedPlan)
+		if err != nil {
+			t.Error(err, "Could not get plan")
+		}
+
+		if returnedPlan.Status.Conditions.IsReady() {
+			log.Info("Plan is ready")
+			close(done)
+		}
+	}
+
+	go wait.Until(statusCheck, time.Second, done)
+
+	select {
+	case <-timeout:
+		t.Errorf("Plan is not ready in time, last status: %v", returnedPlan.Status)
+	case <-done:
+	}
+
+	migration := &v1beta1.Migration{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "Migration",
+			APIVersion: "forklift.konveyor.io/v1beta1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "migration-test",
+			Namespace: namespaceName,
+		},
+		Spec: v1beta1.MigrationSpec{
+			Plan: corev1.ObjectReference{
+				Name:      "plan-test",
+				Namespace: namespaceName,
+			},
+		},
+	}
+
+	log.Info("Creating the migration...")
+	done = make(chan struct{})
+
+	defer cl.Delete(context.TODO(), migration, &client.DeleteOptions{})
+
+	err = cl.Create(context.TODO(), migration, &client.CreateOptions{})
+	if err != nil {
+		t.Fatal(err, "Failed to create the migration")
+	}
+
+	returnedMigration := &v1beta1.Migration{}
+	migrationIdentifier := types.NamespacedName{Namespace: migration.Namespace, Name: migration.Name}
+	// Wait for provider to be ready
+	statusCheck = func() {
+		err = cl.Get(context.TODO(), migrationIdentifier, returnedMigration)
+		if err != nil {
+			t.Error(err, "Could not get migration")
+		}
+
+		if returnedMigration.Status.Conditions.IsReady() {
+			log.Info("Migration is ready")
```

Comment:
In my KinD-tests this is the last log message, nothing visible happens afterwards until the test gets terminated by the test-timeout (even if that is set to 30 minutes).

Reply:
@liranr23 doesn't happen anymore, right?

Reply:
when executing you won't see live logging. if the test passes, you will see `ok` and that's it. in case of failure (error, timeout), you will see the whole logs and the error.

Reply:
ok so based on that, the test doesn't time out anymore for us and we can consider this worksforme